core/src/main/scala/kafka/coordinator/GroupCoordinator.scala

```diff
@@ -681,6 +694,17 @@ class GroupCoordinator(val brokerId: Int,
   def onCompleteJoin(group: GroupMetadata) {
     var delayedStore: Option[DelayedStore] = None
     group synchronized {
+      if (group.generationId == 0) {
```

Comment:
Since we only set `newMemberAdded` if `generationId == 0` in `addMemberAndRebalance`, I feel we do not need to check it here once again?